docs/adr/00xx-file-attachment-rendering.md

```diff
@@ -0,0 +1,121 @@
+Govspeak and re-creating it in Content Publisher. File attachments presented
+in the context of a document would use Govspeak as their means to render,
+whereas those presented outside a document would use Content Publisher
+components.
+
+Pros:
+
+- easy to customise the HTML for Content Publisher needs;
+- able to use the consistent [GOV.UK component approach][govuk-components] to
+  render an attachment.
+
+Cons:
+
+- rendering logic and markup duplicated in two separate repositories
+
+### 3. Both Govspeak and Content Publisher call [GOV.UK Publishing Components][govuk-publishing-components] to render file attachments
+
+This would involve GOV.UK Publishing Components becoming the single place
+used to render file attachments. Govspeak and Content Publisher would both
+individually depend on this gem and call it for file attachment presentation.
+
+Pros:
+
+- single place for file attachment presentation logic;
+- HTML of file attachment can be customised for different scenarios;
+- allows wider use of the component beyond Content Publisher.
+
+Cons:
+
+- most complex approach;
+- additional dependency for Govspeak.
+
+## Decision
+
+We decided that GOV.UK Publishing Components will store the rendering logic
+for presenting File Attachment links and metadata. Govspeak and Content
+Publisher will use these components for their File Attachment rendering.
+
+We felt this approach was the one most consistent with the GOV.UK
+conventions for cross-application component usage. It also allowed us
+flexibility without the risks of duplication.
+
+## Status
+
+Accepted
+
+## Consequences
+
+Govspeak now has a dependency on GOV.UK Publishing Components. In doing so
+this has introduced a [new approach to rendering a Govspeak markdown
+extension][govspeak-approach], which may cause confusion as there were
+already multiple. On the positive side, this will allow other components to be
+rendered through Govspeak which may allow future consolidation of presentation
+logic.
+
+GOV.UK Publishing Components no longer has a dependency on Govspeak. This
+needed to be removed to avoid a cyclic dependency. GOV.UK Publishing Components
+no longer has the means to render Govspeak, this functionality was previously
+used to render markdown and was not used for any specific Govspeak extensions.
+To allow rendering markdown the [Kramdown](https://kramdown.gettalong.org/)
+library is used (which is the same library Govspeak uses).
+
+The rendering of attachments for [publication][publication-schema] documents
+could now be done on the frontend by using the file attachment components.
+This was previously impossible and the rendering of attachments has been
+achieved by Whitehall providing the Publishing API with [an array of HTML
+strings][attachment-html] to represent the attachments. This HTML based approach
+has made it very difficult for teams to extract attachment data from the
+[GOV.UK Content API][govuk-content-api].
```

Comment:
Do we expect that the new file attachment components could make it easier for teams to extract attachment data from Content API?

Reply:
It's somewhat convoluted. With the components there is now the logic to render the components on the frontend so we could stop sending blocks of HTML and replace that with data. When the Content API then had data it'd be easier to extract the information.